src/services/tx/tx-sender/sdk.ts

```diff
@@ -26,6 +26,37 @@ export const getAndValidateSafeSDK = (): Safe => {
   return safeSDK
 }
 
+async function switchOrAddChain(walletProvider: ConnectedWallet['provider'], chainId: string): Promise<void> {
+  const UNKNOWN_CHAIN_ERROR_CODE = 4902
+  const hexChainId = toQuantity(parseInt(chainId))
+
+  try {
+    return await walletProvider.request({
+      method: 'wallet_switchEthereumChain',
+      params: [{ chainId: hexChainId }],
+    })
+  } catch (error) {
+    if ((error as Error & { code: number }).code !== UNKNOWN_CHAIN_ERROR_CODE) {
```

Comment:
Is it possible to use `asError` here?

Reply:
Not much value here, we'd still have to cast it to have a code.